src/shogun/base/DynArray.h

```diff
@@ -441,22 +441,24 @@ template <class T> class DynArray
 			current_num_elements=0;
 		}
 
-		/** randomizes the array (not thread safe!) */
-		void shuffle()
+		T* begin()
 		{
-			for (index_t i=0; i<=current_num_elements-1; ++i)
-				std::swap(
-					array[i], 
-					array[CMath::random(i, current_num_elements - 1)]);
+			return array;
+		}
```

Comment:
I added iterators to ease the shuffling of DynArrays and DynamicObjectArrays. Not sure if this is ok